core/unix/os.c

```diff
@@ -1364,7 +1380,6 @@ os_slow_exit(void)
 
     if (doing_detach) {
         vsyscall_page_start = NULL;
-        IF_DEBUG(num_fd_add_pre_heap = 0;)
```

Comment:
So this should be back to 0 and that's why this isn't needed?  Maybe add an assert to make this clearer.

Reply:
Done.